### ISSUE TITLE
Extend the API with missing functions from what was present in box2d

### ISSUE THREAD

**Issue by Ughuuu** [erincatto/box2c#111]
Starting with my needs for [godot-box2d](https://github.com/appsinacup/godot-box2d) which integrates box2d into godot. Godot has its own physics system that has an API that needs implementing. With box2d this was possible, but in box2c some functions are missing.

```
- b2Body_SetFixedRotation() // be able to set fixed rotation of body after it's created.
- b2Body_ForceSleep() // force a body to sleep. I used to call this after making a body static. Not 100% sure if it was needed, but nice to have since there is a wake function as well.
- b2Body_GetBullet() // get if a body has ccd enabled
- b2Body_SetBullet() // set if a body has ccd enabled
- b2Body_IsSleepingAllowed() // get if a body is allowed to sleep
- b2World_GetBodies() // some function to get bodies in the world. I use this to go through all bodies and check which of them moved so I can call a callback on them and mark them as active bodies. There might be a better way to do it, not sure.
```

Haven't went yet into setting callbacks and other things, still wip, but these are just a few. I will add more as I advance.

**Comment by erincatto** [comment]
Static bodies always sleep.

Version 3 has persistent islands and bodies no longer have an individual sleep state. Instead the whole island is awake or asleep. Waking a body is simple. It wakes the whole island. On the other hand it might be confusing that putting a body to sleep would require all bodies in the island to also be put to sleep.

**Comment by erincatto** [comment]
Iterating over all bodies in the world will result in per performance. I have added this function to indicate which bodies have moved.
```c
/// Get the body events for the current time step. The event data is transient. Do not store a reference to this data.
B2_API b2BodyEvents b2World_GetBodyEvents(b2WorldId worldId);
```

**Comment by erincatto** [comment]
Also I already added this function:
```c
/// @return is sleeping enabled for this body?
B2_API bool b2Body_IsSleepEnabled(b2BodyId bodyId);
```

The bullet feature is in the v3.0 task list #44

**Comment by erincatto** [comment]
These are all addressed in #120. The bullet task will come later as part of #44.
